tests/components/knx/test_diagnostic.py

```diff
@@ -0,0 +1,63 @@
+"""Tests for the diagnostics data provided by the KNX integration."""
+from unittest.mock import patch
+
+from aiohttp import ClientSession
+
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.components.knx.conftest import KNXTestKit
+
+
+async def test_diagnostics(
+    hass: HomeAssistant,
+    hass_client: ClientSession,
+    mock_config_entry: MockConfigEntry,
+    knx: KNXTestKit,
+):
+    """Test diagnostics."""
+    await knx.setup_integration({})
+
+    with patch("homeassistant.config.async_hass_config_yaml", return_value={}):
+        assert await get_diagnostics_for_config_entry(
+            hass, hass_client, mock_config_entry
+        ) == {
+            "config_entry_data": {
+                "connection_type": "automatic",
+                "individual_address": "15.15.250",
+                "multicast_group": "224.0.23.12",
+                "multicast_port": 3671,
+            },
+            "configuration_error": None,
+            "configuration_yaml": None,
+            "xknx": {"current_address": "0.0.0", "version": "0.19.0"},
```

Comment:
can we somehow extract the version number from the manifest to avoid having to update this test on every xknx update?

Reply:
We can also just verify that it is a string so the tests also pass when we use a .dev version of xknx. 
Or mock `xknx.__version__.py`

Reply:
AFAIK we can't mock `xknx.__version__.py`. I have now just overwritten the version for this test case so we don't need to bother with it. I don't think we should parse a file in a test.